File: python/params/PARAMS_ion_acoustic_waves.py
import jax.numpy as jnp

### General parameters ###
mycase         = "ion_accoustic_waves"     # "two_stream"
Nx             = 2**8                      # Number of spatial grid points
Nv             = 2**9                      # Number of velocity grid points
Ns             = 2                         # Number of species (electrons and ions)
method         = "NuFi"
species_name   = ["electrons", "ions"]     # Name of the different species
Mr             = 1000                      # Mass ratio for ions
Mass           = [1, Mr]                   # Mass of species
charge         = [-1, 1]                   # Charge of species
Nt_max         = 4000                      # Maximum number of time steps
dt             = 1/4                       # Time step size
dt_save        = 100                       # Save after dt_save time
Tend           = 500                       # End time of simulation

### Initial condition parameters ###
k      = 0.5                               # Wave number
alpha  = 0.5                               # Perturbation amplitude
pert   = lambda x: alpha * jnp.cos(k * x)  # Perturbation function

### Electrons ###
Ue     = -2                                                                                 # Electron drift velocity
fe0    = lambda x, v: (1 + pert(x)) / (jnp.sqrt(2 * jnp.pi)) * jnp.exp(-(v - Ue) ** 2 / 2)  # Electron distribution
fi0    = lambda x, v: jnp.sqrt(Mr / (2 * jnp.pi)) * jnp.exp(-Mr * v ** 2 / 2)               # Ion distribution
fini   = [fe0, fi0]

### Grid parameters ###
Lx     = 2 * jnp.pi / k                    # Spatial domain length
Lv_s   = [8, 0.1 * jnp.pi]                 # Velocity domain lengths for each species

File: Benchmark/_Matlab_Version_Params/PARAMS_ion_acoustic_waves.m
params = struct();
params.mycase = "ion_accoustic_waves_weak";          % "two_stream"
params.Nx = 2^9;                            % Number of spatial grid points
params.Nv = 2^9;                            % Number of velocity grid points
params.Ns = 2;                              % Number of species (electrons and ions)
params.method="NuFI";
params.species_name = ["electrons","ions"]; % name of the different species
params.Mr = 1000;                           % Mass ratio for ions
params.Mass = [1, params.Mr];               % Mass of species
params.charge = [-1, 1];                    % Charge of species
params.Nt_max = 5000;                       % Maximum number of time steps
params.dt = 10;                            % Time step size
params.Nsubs = 8;
params.dt_save = params.dt;                       % Save after dt_save time
params.Tend = 5000;                          % End time of simulation

% Initial condition parameters
params.k = 0.5;                             % Wave number
params.alpha = 0.01;                         % Perturbation amplitude
params.pert = @(x) params.alpha * cos(params.k * x); % Perturbation function

% Electrons
params.Ue = -2;                   % Electron drift velocity
params.fe0 = @(x, v) (1+params.pert(x)) ./ (sqrt(2 * pi)) .* (exp(-(v - params.Ue).^2 / 2)); % Electron distribution
params.fi0 = @(x, v) sqrt(params.Mr / (2 * pi)) .* (exp(-params.Mr * (v).^2 / 2)); % Ion distribution
params.fini = {params.fe0,params.fi0};
% Grid parameters
params.Lx = 2 * pi / params.k;    % Spatial domain length
params.Lv_s = [8, 0.1 * pi]; % Velocity domain lengths for each species
